src/app/route-history/route-history.service.ts

```diff
@@ -11,12 +11,19 @@ export class RouteHistoryService implements OnDestroy {
   public previousRoute: string;
 
   private subscription: Subscription;
+  private popstate: boolean = false;
 
   constructor(router: Router) {
     this.subscription = router.events.subscribe((event: Event) => {
+      if (event instanceof NavigationStart) {
+        if (event.navigationTrigger === 'popstate') {
```

Comment:
Would it make sense to combine this into a single if?

On the flip side, if the intention of the nested if is to clearly signal that this function is serving a *variety* of distinct event handlers for different event types (and you're nesting inside), that might be even clearer by creating named handlers (e.g. `handleNavigationStart(event)` and `handleNavigationEnd(end)`) and this handler serves as only a router that calls the more specific event handlers.

All of this is purely for your consideration of course